packages/lisk-transactions/src/transactions/5_dapp_transaction.ts

```diff
@@ -0,0 +1,510 @@
+							error.dataPath,
+						),
+			  )
+			: [];
+		if (!typeValid) {
+			throw new TransactionMultiError('Invalid field types.', tx.id, errors);
+		}
+		this.asset = tx.asset as DappAsset;
+	}
+
+	public static create(input: CreateDappInput): object {
+		const { passphrase, secondPassphrase, options } = input;
+
+		const transaction = {
+			...createBaseTransaction(input),
+			type: 5,
+			fee: DAPP_FEE.toString(),
+			asset: {
+				dapp: options,
+			},
+		};
+
+		if (!passphrase) {
+			return transaction;
+		}
+
+		const transactionWithSenderInfo = {
+			...transaction,
+			senderId: transaction.senderId as string,
+			senderPublicKey: transaction.senderPublicKey as string,
+			recipientId: transaction.senderId as string,
+		};
+
+		const dappTransaction = new DappTransaction(transactionWithSenderInfo);
+		dappTransaction.sign(passphrase, secondPassphrase);
+
+		return dappTransaction.toJSON();
+	}
+
+	public static fromJSON(tx: TransactionJSON): DappTransaction {
+		const transaction = new DappTransaction(tx);
+		const { errors, status } = transaction.validateSchema();
+
+		if (status === Status.FAIL && errors.length !== 0) {
+			throw new TransactionMultiError(
+				'Failed to validate schema.',
+				tx.id,
+				errors,
+			);
+		}
+
+		return transaction;
+	}
+
+	protected getAssetBytes(): Buffer {
+		const DAPP_TYPE_LENGTH = 4;
+		const DAPP_CATEGORY_LENGTH = 4;
+		const {
+			name,
+			description,
+			tags,
+			link,
+			icon,
+			type,
+			category,
+		} = this.asset.dapp;
+		const nameBuffer = Buffer.from(name, 'utf8');
+		const linkBuffer = link ? Buffer.from(link, 'utf8') : Buffer.alloc(0);
+		const typeBuffer = Buffer.alloc(DAPP_TYPE_LENGTH);
+		typeBuffer.writeIntLE(type, 0, DAPP_TYPE_LENGTH);
+		const categoryBuffer = Buffer.alloc(DAPP_CATEGORY_LENGTH);
+		categoryBuffer.writeIntLE(category, 0, DAPP_CATEGORY_LENGTH);
+
+		const descriptionBuffer = description
+			? Buffer.from(description, 'utf8')
+			: Buffer.alloc(0);
+		const tagsBuffer = tags ? Buffer.from(tags, 'utf8') : Buffer.alloc(0);
+		const iconBuffer = icon ? Buffer.from(icon, 'utf8') : Buffer.alloc(0);
+
+		return Buffer.concat([
+			nameBuffer,
+			descriptionBuffer,
+			tagsBuffer,
+			linkBuffer,
+			iconBuffer,
+			typeBuffer,
+			categoryBuffer,
+		]);
+	}
+
+	public assetToJSON(): object {
+		return {
+			...this.asset,
+		};
+	}
+
+	public getRequiredAttributes(): Attributes {
+		const attr = super.getRequiredAttributes();
+		const filterObject = {
+			dappName: [this.asset.dapp.name],
+		};
+		const uniqueFields = this.asset.dapp.link
+			? {
+					...filterObject,
+					dappLink: [this.asset.dapp.link],
+			  }
+			: filterObject;
+
+		return {
+			...attr,
+			[ENTITY_TRANSACTION]: uniqueFields,
+		};
+	}
+
+	public verifyAgainstOtherTransactions(
+		transactions: ReadonlyArray<TransactionJSON>,
+	): TransactionResponse {
+		const sameTypeTransactions = transactions
+			.filter(tx => tx.type === this.type)
+			.map(tx => new DappTransaction(tx));
```

Comment:
necessary to initialize here? at this stage will `transactions` already be type checked?